### PR TITLE
chore: fix all eslint warnings and turn them into errors

### PR DESCRIPTION
**Pull request by bjornstar**
### Why

When we run `eslint` we get a number of warnings:

```
❯ npm run eslint

> @react-three/drei@0.0.0-semantic-release eslint
> eslint --fix {src,.storybook}/**/*.{ts,tsx}


/Users/bjorn.stromberg/drei/.storybook/stories/Stage.stories.tsx
  2:54  warning  'array' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/Users/bjorn.stromberg/drei/src/core/BakeShadows.tsx
  12:6  warning  React Hook React.useEffect has a missing dependency: 'gl.shadowMap'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/Bounds.tsx
  156:6   warning  React Hook React.useMemo has missing dependencies: 'current' and 'goal'. Either include them or remove the dependency array          react-hooks/exhaustive-deps
  165:25  warning  'event' is defined but never used. Allowed unused args must match /^_/u                                                              @typescript-eslint/no-unused-vars
  169:6   warning  React Hook React.useLayoutEffect has missing dependencies: 'clip' and 'current'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/GizmoViewcube.tsx
  82:6  warning  React Hook React.useMemo has a missing dependency: 'faces'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/GradientTexture.tsx
  27:6  warning  React Hook React.useMemo has missing dependencies: 'colors' and 'size'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/Instances.tsx
   41:49  warning  React Hook React.useLayoutEffect has a missing dependency: 'subscribe'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
   75:8   warning  React Hook React.useLayoutEffect has a missing dependency: 'limit'. Either include it or remove the dependency array      react-hooks/exhaustive-deps
   83:15  warning  'state' is defined but never used. Allowed unused args must match /^_/u                                                   @typescript-eslint/no-unused-vars
  121:8   warning  React Hook React.useMemo has an unnecessary dependency: 'children'. Either exclude it or remove the dependency array      react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/OrthographicCamera.tsx
  23:6  warning  React Hook React.useLayoutEffect has a missing dependency: 'manual'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/PerspectiveCamera.tsx
  24:6  warning  React Hook React.useLayoutEffect has a missing dependency: 'manual'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/Points.tsx
   37:8   warning  React Hook React.useLayoutEffect has a missing dependency: 'limit'. Either include it or remove the dependency array      react-hooks/exhaustive-deps
   85:8   warning  React Hook React.useMemo has an unnecessary dependency: 'children'. Either exclude it or remove the dependency array      react-hooks/exhaustive-deps
  132:49  warning  React Hook React.useLayoutEffect has a missing dependency: 'subscribe'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/PositionalAudio.tsx
  27:8  warning  React Hook React.useEffect has a missing dependency: 'autoplay'. Either include it or remove the dependency array                 react-hooks/exhaustive-deps
  39:8  warning  React Hook React.useEffect has missing dependencies: 'camera' and 'listener'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/Segments.tsx
    5:53  warning  'LineMaterialParameters' is defined but never used. Allowed unused vars must match /^_/u                            @typescript-eslint/no-unused-vars
  112:51  warning  React Hook React.useLayoutEffect has a missing dependency: 'api'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/TrackballControls.tsx
   1:10  warning  'EventManager' is defined but never used. Allowed unused vars must match /^_/u                                     @typescript-eslint/no-unused-vars
  54:8   warning  React Hook React.useEffect has a missing dependency: 'controls'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/core/useDetectGPU.tsx
  6:10  warning  'GPUTier' is assigned a value but never used. Allowed unused vars must match /^_/u     @typescript-eslint/no-unused-vars
  6:19  warning  'setGPUTier' is assigned a value but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

/Users/bjorn.stromberg/drei/src/core/useIntersect.tsx
  30:6  warning  React Hook React.useEffect has a missing dependency: 'onChange'. Either include it or remove the dependency array. If 'onChange' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/web/CycleRaycast.tsx
  97:6  warning  React Hook React.useEffect has missing dependencies: 'onChanged' and 'portal'. Either include them or remove the dependency array. If 'onChanged' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/web/ScrollControls.tsx
   86:6   warning  React Hook React.useMemo has missing dependencies: 'el', 'fill', and 'fixed'. Either include them or remove the dependency array                     react-hooks/exhaustive-deps
  127:6   warning  React Hook React.useEffect has missing dependencies: 'events', 'gl.domElement', and 'raycaster'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
  138:23  warning  'e' is defined but never used. Allowed unused args must match /^_/u                                                                                  @typescript-eslint/no-unused-vars
  171:6   warning  React Hook React.useEffect has a missing dependency: 'enabled'. Either include it or remove the dependency array                                     react-hooks/exhaustive-deps

/Users/bjorn.stromberg/drei/src/web/useCursor.tsx
  9:6  warning  React Hook React.useEffect has missing dependencies: 'onPointerOut' and 'onPointerOver'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

✖ 31 problems (0 errors, 31 warnings)
```

### What

Applied fixes for all of the warnings given by `eslint`

### Checklist

- [x] Ready to be merged

### CODE REVIEW

**Review by gsimone — APPROVED** [review]

File: src/core/Bounds.tsx [hunk 1]
```diff
@@ -160,7 +160,7 @@ export function Bounds({ children, damping = 6, fit, clip, margin = 1.2, eps = 0
         return this
       },
     }
-  }, [box, camera, controls, margin, damping, invalidate])
+  }, [box, camera, controls, current, damping, goal, invalidate, margin])
```
Comment (gsimone) [inline src/core/Bounds.tsx:163]: @drcmda do these matter or were they missing by mistake?

File: src/core/Bounds.tsx [hunk 2]
```diff
@@ -169,12 +169,11 @@ export function Bounds({ children, damping = 6, fit, clip, margin = 1.2, eps = 0
...
     if (controls) {
       // Try to prevent drag hijacking
-      const callback = (event) => (current.animating = false)
+      const callback = () => (current.animating = false)
       controls.addEventListener('start', callback)
       return () => controls.removeEventListener('start', callback)
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [clip, fit, controls])
+  }, [api, clip, controls, current, fit])
```
Comment (gsimone) [inline src/core/Bounds.tsx:177]: @drcmda  Same as the prev. comment

**Review by gsimone — CHANGES_REQUESTED** [review]
I'm dubious regarding the changes in useEffect dependencies, I'd eslint-disable the checks for the existing ones instead of introducing possible changes. 

We can then have a separate task to fix & test them.

**Comment by bjornstar** [comment]
> I'm dubious regarding the changes in useEffect dependencies, I'd eslint-disable the checks for the existing ones instead of introducing possible changes.
> 
> We can then have a separate task to fix & test them.

If we don't care about the hook dependency warnings, we can turn that rule off.

It's up to you, but I feel like having a bunch of `@eslint-disable` comments and warnings is very noisy.

**Comment by bjornstar** [comment]
I went ahead and backed out the hook dependency changes and turned off that lint rule.

**Comment by gsimone** [comment]
Let's go with this for now, we'll tackle the dependencies in another PR

**Pull request merged**